### PR TITLE
Implemented orchestrator NG handler

### PR DESCRIPTION
**Pull request by victorskl**
* NG handlers are centered around `portal_run_id`
  and workflow updates are mapped from external event
  into internal domain event representation
* Added WorkflowRunStateChange JSON schema
  that schematise domain event construct
* Added Pydantic for handling autogen and model construct
* Added SQS Batch Event router Lambda
  and hooking it to oncoanalyser Nextflow stack job queue
* Added unit test cases and doc

### CODE REVIEW

**Comment by victorskl** [comment]
[Portal Release - Ocicat: Implement Portal automation integration for oncoanalyser](https://trello.com/c/sZFv7FhZ/1488-portal-release-ocicat-implement-portal-automation-integration-for-oncoanalyser)

File: data_processors/pipeline/services/workflow_srv.py [hunk 1]
```diff
@@ -59,38 +55,66 @@ def create_or_update_workflow(model: dict):
...
-
         _end = model.get('end')
         if _end:
             if isinstance(_end, datetime):
                 workflow.end = _end if is_aware(_end) else make_aware(_end)
             else:
                 workflow.end = _end  # expect end in raw zone-aware UTC datetime string e.g. "2020-06-25T10:45:20.438Z"
 
-        workflow.notified = model.get('notified')
+    # --- business logic: below are mutable fields regardless of new or existing records, if they have been payload
```
Comment (reisingerf) [inline data_processors/pipeline/services/workflow_srv.py:82]: Mutable yes, but ideally only set once then should match anything that follows, right?
(could be future improvement)

File: data_processors/pipeline/domain/event/wrsc.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from typing import Optional
+
+from pydantic import BaseModel, Field, AliasChoices
+
+
+class WorkflowRunStateChange(BaseModel):
+    portal_run_id: str
+    type_name: str
+    version: str
+    output: str
```
Comment (reisingerf) [inline data_processors/pipeline/domain/event/wrsc.py:17]: No input?
Because it's not used for initial workflow creation, i.e. only follow-on events?

**Review by reisingerf — APPROVED** [review]

File: data_processors/pipeline/services/workflow_srv.py (hunk 1 above)
Comment (victorskl) [reply data_processors/pipeline/services/workflow_srv.py:82]: Okay! Noted well. Let us observe and improve it in next iteration.

**Comment by victorskl** [comment]
Merging...

**Pull request merged**